src/class.c

```diff
@@ -1634,22 +1661,27 @@ bool InterfaceDeclaration::isBaseOf(ClassDeclaration *cd, int *poffset)
     {
         BaseClass *b = cd->interfaces[j];
 
-        //printf("\tbase %s\n", b->base->toChars());
-        if (this == b->base)
+        //printf("\tX base %s\n", b->sym->toChars());
+        if (this == b->sym)
         {
             //printf("\tfound at offset %d\n", b->offset);
             if (poffset)
             {
                 *poffset = b->offset;
-                if (j && cd->isInterfaceDeclaration())
+                // The conversion is interface(cd) to interface(this).
+                if (cd->isInterfaceDeclaration())
```

Comment:
I meant to use (1) for the pointer comparison. I don't want _d_interface_cast or _d_dynamic_cast to be invoked when doing a conversion to a base interface:
```
interface I {}
interface J {}
interface K : I, J {}

class C : K {}
class D : L {}

void main()
{
	C c = new C;
	K k = c;
	I ic = c;
	I ik = k;
	J jc = c;
	J jk = k;
}
```
The assignments of `ic` and `ik` are done by adding a pointer now, but `ik` would require a dynamic cast runtime call with this PR. For `jc` and `jk` the compiler currently inserts _d_dynamic_cast and _d_interface_cast, respectively. AFAIU this PR removes the dynamic cast for `jc`, that's good.
Regarding the cast K -> J, I'm beginning to understand the need to do dynamic_casting, though I believe that C++ doesn't have to do it with virtual inheritence. (Not sure these are exactly the same things, though.)

Considering the first base class interface:
```
interface I {}
interface M : I {}
class D : M {}

void main()
{
	D d = new D;
	M m = d;
	I i = m;
}
```
the vtbl for M completely contains the interface for I, so a dynamic cast should be unnecessary here.